hack/generator/pkg/astmodel/resource_type.go

```diff
@@ -24,6 +24,7 @@ type ResourceType struct {
 	status           Type
 	isStorageVersion bool
 	owner            *TypeName
+	properties       map[PropertyName]*PropertyDefinition
```

Comment:
@theunrepentantgeek - Sorry I missed looking at this PR on my Friday, but why do we need this change for the storage versions? The property doesn't have to be on the top level resource type I don't think. It really should be in spec in the storage version too, shouldn't it?

Maybe we can sync up offline or in the meeting today?